### ISSUE TITLE
Missing docstrings

### ISSUE THREAD

**Issue by fredrikaverpil** [mherrmann/fbs#15]
I feel that it's difficult to follow the code of fbs since a lot of docstrings are missing. For example, there's no docstring on the ApplicationContext class.

I also have a hard time following why the application won't run if I remove the @cached_property from the application_context.py. I'm somewhat familiar with LRU cache, but not to the extent that I understand why your're using it in application_context.py.

In short; docstrings would help when digging.

**Comment by mherrmann** [comment]
Good point, I can totally see why more explaining comments would be useful. I'll keep it in mind, thanks.

**Comment by mherrmann** [comment]
I now added docstrings to all (public) functions. Also, I wrote a [Manual](https://build-system.fman.io/manual/) that goes into more technical detail than the tutorial. Hopefully, it will let you (and others) avoid having to dig through the details of the source code to find out what's going on.

**Comment by fredrikaverpil** [comment]
Excellent, thank you!
